polclassifier/ml_logic/preprocessing.py

```diff
@@ -160,3 +161,31 @@ def save_processed_to_cache(
     X.to_csv(X_path, index=False)
     y.to_csv(y_path, index=False)
     return
+
+
+def embed_and_pad(X, embedding, stop_words=stop_words):
+    X_embed = embed_sentences(X, embedding, stop_words)
+    
+    maxlen = max([len(x) for x in X_embed])
+    X_pad = pad_sequences(X_embed, dtype='float32', padding='post', maxlen=maxlen)
+
+
+
+# Function to convert a sentence (list of words) into a matrix representing the words in the embedding space
+def embed_sentence_with_TF(sentence, embedding, stop_words=stop_words):
+    embedded_sentence = []
+    for word in sentence:
+        if word in embedding and word not in stop_words:
+            embedded_sentence.append(embedding[word])
+        
+    return np.array(embedded_sentence)
+
+# Function that converts a list of sentences into a list of matrices
+def embed_sentences(embedding, sentences):
+    embed = []
+    
+    for sentence in sentences:
+        embedded_sentence = embed_sentence_with_TF(embedding, sentence.split())
```

Comment:
Is the argument for stop_words necessary?

Reply:
Thanks, fixed!